tools/CopyrightDateTool.scala

```diff
@@ -0,0 +1,171 @@
+import sys.process._
+
+object CopyrightDateTool {
+  val usage = """
+
+    This tool compares a copyright date in files with the date of last change in git.
+    If the message of last change contains "[Sync copyright dates]" it will be ignored and 
+    we will take next commit.
+
+    analyze - just shows if anything is ok
+    update - file fixing
+
+    From root folder:
+    Usage: scala tools/CheckDates.scala analyze
+           scala tools/CheckDates.scala update
+
+  """
+  trait DateComparison {
+    def fileLastUpdateYear: String
+    def gitLastUpdateYear: String
+    def fileCreationYear: String
+    def gitCreationYear: String
+
+    def isUpdateDateWrong(): Boolean = {
+      fileLastUpdateYear.toInt != gitLastUpdateYear.toInt
+    }
+
+    def isUpdateDateWrongStr(): Boolean = {
+      fileLastUpdateYear != gitLastUpdateYear
+    }
+  }
+
+  case class FileDates(pathToFile: String,
+                      fileLastUpdateYear: String,
+                      gitLastUpdateYear: String,
+                      fileCreationYear: String,
+                      gitCreationYear: String) extends DateComparison
+
+  case class GitInfo(message: String, date: String)
+
+  def readFilesList(): Array[String] = {
+    val git_files = Seq("/bin/sh", "-c", 
```

Comment:
https://github.com/scala/scala/pull/5145#discussion-diff-63394600R42

I answered already there. There is no huge difference. I chose `Seq()` because it works exactly like regular shell(no `#|` etc.)